### PR TITLE
Added lint-self-closing-void-elements rule

### PR DESCRIPTION
**Pull request by oriSomething**
related #77

tell me if there is anything to fix

### CODE REVIEW

File: ext/plugins/lint-self-closing-void-elements.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return this.config && node.type === 'ElementNode' && VOID_TAGS[node.tag];
+  };
+
+  LogSelfClosingVoidElements.prototype.process = function(node) {
+    var source = this.sourceForNode(node).trim();
+    var sourceEndTwoCharacters = source.slice(source.length - 2);
+
+    if (sourceEndTwoCharacters === '/>') {
+      var warning = 'Self closing void element as <' + node.tag +
+        '> is redundant ' + calculateLocationDisplay(this.options.moduleName, node.loc.start);
```
Comment (rwjblue) [inline ext/plugins/lint-self-closing-void-elements.js]: Can you suggest the fix also in the comment?

**Comment by rwjblue** [comment]
* Can you link to the spec listing of void tags above the VOID_TAGS constant?
* Can you link to some reasoning for the rule (preferably spec language or something)? I just know I'll be asked "what is wrong with <br/>?!?" And it would be nice to just have a handy link in the README...

**Comment by oriSomething** [comment]
before i'm adding to the code:
- [the specs of void elements](https://www.w3.org/TR/html-markup/syntax.html#void-element)
- in the spec it's written it's an optionally. i can't find a specific link why it's a bad thing (although i remember they made this options to better compatible with XHTML). the reason i think it's bad because: 1) it's not XML (or variant SVG / XHTML); 2) it's like adding a extract space `<hr    >` it's just redundant and unneeded but it is optionally

**Comment by rwjblue** [comment]
Kk, sounds good.

**Comment by oriSomething** [comment]
i've updated to readme to something clearer i hope

**Comment by olleolleolle** [comment]
Cool rule!

Spelling note: The HTML5 spec spells it `self-closing`, with a hyphen. Example: https://www.w3.org/TR/html/syntax.html#elements-0

**Comment by rwjblue** [comment]
Great catch! 

@oriSomething - Can you update that? Sorry for letting this linger, it's ready to land once the verbiage is tweaked...

**Comment by oriSomething** [comment]
@rwjblue done!

**Pull request merged**